### PR TITLE
Rate limit with keycard

### PR DESCRIPTION
**Pull request by ChaituVR**
Fixes https://github.com/snapshot-labs/score-api/pull/483

🔴  Need a new version from keycard.js

Changes:
- Same implementation as snapshot-hub to restrict based on IP and API key with key card

```[tasklist]
### Admin Tasks
- [x] Need to add `KEYCARD_URL` and `KEYCARD_SECRET` to env on DO
- [x] Need to add `SCORE_API_URL` in sequencer DO `https://score.snapshot.org?apiKey=...`
- [x] Need to add `SCORE_API_URL` in hub DO `https://score.snapshot.org?apiKey=...`
```

### CODE REVIEW

**Review by wa0x6e — COMMENTED** [review]
I am getting the express rate limit headers twice
```
curl -IL http://localhost:3003/api/strategies
HTTP/1.1 200 OK
X-Powered-By: Express
Access-Control-Allow-Origin: *
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 59
Date: Mon, 07 Aug 2023 13:29:02 GMT
X-RateLimit-Reset: 1691414947
RateLimit-Limit: 60
RateLimit-Remaining: 59
RateLimit-Reset: 5
Content-Type: application/json; charset=utf-8
Content-Length: 686335
ETag: W/"a78ff-W00Dkrou5tnc6ccr9SrW1NYqhc8"
Connection: keep-alive
Keep-Alive: timeout=5
```

Is that normal ?

And should we have the same header name for api rate limit and express rate limit, or is it better to have separate name ?

**Comment by ChaituVR** [comment]
`X-RateLimit-` and `RateLimit-` both are coming from express-rate-limit

File: src/helpers/rateLimit.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import rateLimit from 'express-rate-limit';
+import { getIp, rpcError } from '../utils';
+
+export default rateLimit({
+  windowMs: 20 * 1e3,
+  max: 60,
+  keyGenerator: (req) => getIp(req),
+  standardHeaders: true,
```
Comment (ChaituVR) [inline src/helpers/rateLimit.ts:8]:
```suggestion
```

**Review by wa0x6e — APPROVED** [review]
tAck

File: src/helpers/rateLimit.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import rateLimit from 'express-rate-limit';
+import { getIp, rpcError } from '../utils';
+
+export default rateLimit({
+  windowMs: 20 * 1e3,
+  max: 60,
+  keyGenerator: (req) => getIp(req),
```
Comment (ChaituVR) [inline src/helpers/rateLimit.ts:7]:
```suggestion
  standardHeaders: true,
  legacyHeaders: false,
  keyGenerator: (req) => getIp(req),
```

File: package.json [hunk 3]
```diff
@@ -39,7 +41,7 @@
     "ts-jest": "^28.0.0"
   },
   "engines": {
-    "node": ">=16.0.0 <17.0.0"
+    "node": ">=16.0.0 <=18.16.1"
```
Comment (wa0x6e) [inline package.json:44]: Don't we want to exclude node 17, which is end of life ?

Comment (ChaituVR) [reply package.json:44]:
```suggestion
    "node": ">=16.0.0 <17.0.0"
```

**Pull request merged**